cdap-proto/src/main/java/co/cask/cdap/proto/id/ElementId.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.proto.id;
+
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.element.ElementType;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+
+import java.util.Iterator;
+import java.util.Objects;
+
+/**
+ * Uniquely identifies a particular instance of an element.
```

Comment:
Can you add some instructions on how to add a new element type? (it requires that ElementType enum is updated).